### DISCUSSION TITLE
Multiple event triggers with config

### DISCUSSION THREAD

**Discussion by amannn** [community/community#25595] [Actions]
<p>I want to run a given workflow both on push and based on a cronjob.</p>
<p>I assumed this to work:</p>
<pre><code>on:
  - push
  - schedule:
      - cron: '0 0 * * *' # Once per day at midnight
</code></pre>
<p>However that throws with:</p>
<pre><code>Invalid type for on
</code></pre>
<p>How can I get this to work?</p>

**Comment by chrispat** [comment, marked as answer]
<p>When you want to specify properties for the events you need to specify as a map instead of an array.</p>
<pre><code>on:
  push
  schedule:
      - cron: '0 0 * * *' # Once per day at midnight
</code></pre>

**Comment by tedmiston** [comment]
<p>When I try the accepted solution, the workflow editor seems to think the syntax is invalid [for having a string inside the yaml mapping vs a key with a trailing colon] (see screenshot).</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/c/cb3f299fc7fc2f77f350f62049b33376777a725a.png" data-download-href="/uploads/short-url/t00fBFAp9xFDLK8rQni3TcZeDQ6.png?dl=1" title="Screenshot 2019-10-24 at 8.07.58 PM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/366688/181091946-49ceb3e4-73cc-4b29-9354-c624fa7f6da6.png" alt="Screenshot 2019-10-24 at 8.07.58 PM.png" data-base62-sha1="t00fBFAp9xFDLK8rQni3TcZeDQ6" width="690" height="355" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/c/cb3f299fc7fc2f77f350f62049b33376777a725a_2_690x355.png, https://aws1.discourse-cdn.com/github/original/2X/c/cb3f299fc7fc2f77f350f62049b33376777a725a.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/c/cb3f299fc7fc2f77f350f62049b33376777a725a.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/c/cb3f299fc7fc2f77f350f62049b33376777a725a_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2019-10-24 at 8.07.58 PM.png</span><span class="informations">990×510 54.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Is it possible that push here should instead be push:?</p>
<p>I’ve tried using it with the colon in a quick test just now and both the push run and the schedule run executed as expected.</p>

**Comment by elgohr** [comment]
<p>You can use</p>
<pre><code>on:
  push:
    branches:
      - master

  schedule:
    - cron: '*/5 * * * *'
</code></pre>

**Comment by riker09** [comment]
<p>Even though I’m a little late on this one, but yes</p>
<pre><code>on:
  push:
  schedule:
    - cron: '0 0 * * * '
</code></pre>
<p>is perfectly valid YAML. It’s all about the nesting. :slight_smile:</p>

**Reply by tomchambers2** [reply]
This is the only syntax that has worked for me

**Comment by streamnsight** [comment]
but something like 
```yaml
on:
  workflow_dispatch:
  schedule:
  - cron: '0 0 * * 1'
```
Does not work, and neither does:
```yaml
on:
  workflow_dispatch
  schedule:
  - cron: '0 0 * * 1'
```

any suggestion?

**Reply by chrispat** [reply]
That works fine https://github.com/chrispat/chrispat-action/blob/master/.github/workflows/use-action.yml.

**Reply by jasonm23** [reply]
I think you need to indent the `- cron: ...`

**Reply by riker09** [reply]
I think the extra-identation is not required.

```yaml
on:
  schedule:
  - cron: "0 0 * * 1"
```

is the same as

```yaml
on:
  schedule:
    - cron: "0 0 * * 1"
```

and should be the same as:

```yaml
on:
  schedule: [{ "cron": "0 0 * * 1" }]
```

YAML can be translated into JSON and vice-versa. (to some degree)

The example above without the colon definitely does not work. I'm not sure why this

```yaml
on:
  workflow_dispatch:
  schedule:
  - cron: '0 0 * * 1'
```

does not work. `workflow_dispatch` is one trigger, `schedule` another. Totally independent of each other. `schedule` takes multiple arguments here, one line per arg. Disclaimer: I haven't used GitHub Actions in over two years (not by choice) so maybe they have changed how `schedule` works in the meantime.

**Reply by streamnsight** [reply]
sorry I never replied that I got this working. My actual cron definition had an issue and github action just failed to understand it, but didn't explain why, and I thought it was related to the multiple targets syntax, but it was really the cron job not being understood.
Thanks for your comments though.
